contracts/HubPool.sol

```diff
@@ -472,6 +498,41 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         );
     }
 
+    function _exchangeRateCurrent(address l1Token) internal returns (uint256) {
+        PooledToken storage pooledToken = pooledTokens[l1Token]; // Note this is storage so the state can be modified.
+        uint256 lpTokenTotalSupply = IERC20(pooledToken.lpToken).totalSupply();
+        if (lpTokenTotalSupply == 0) return 1e18; // initial rate is 1 pre any mint action.
+
+        // First, update fee counters and local accounting of finalized transfers from L2 -> L1.
+        _updateAccumulatedLpFees(pooledToken); // Accumulate all allocated fees from the last time this method was called.
+        // _sync(); // Fetch any balance changes due to token bridging finalization and factor them in.
+
+        // ExchangeRate := (liquidReserves + utilizedReserves - undistributedLpFees) / lpTokenSupply
+        // Note that utilizedReserves can be negative. If this is the case, then liquidReserves is offset by an equal
+        // and opposite size. LiquidReserves + utilizedReserves will always be larger than undistributedLpFees so this
+        // int will always be positive so there is no risk in underflow in type casting in the return line.
+        int256 numerator = int256(pooledToken.liquidReserves) +
+            pooledToken.utilizedReserves -
+            int256(pooledToken.undistributedLpFees);
+        return (uint256(numerator) * 1e18) / lpTokenTotalSupply;
+    }
+
+    // Update internal fee counters by adding in any accumulated fees from the last time this logic was called.
+    function _updateAccumulatedLpFees(PooledToken storage pooledToken) internal {
+        uint256 accumulatedFees = _getAccumulatedFees(pooledToken.undistributedLpFees, pooledToken.lastLpFeeUpdate);
+        pooledToken.undistributedLpFees -= accumulatedFees;
```

Comment:
Should this be a += instead of a -=?

Reply:
its weird that we fetch all accumulated fees in the contract and then subtract that amoutn from the undistributed LP fees. I would have instinctively guessed that "undistributed fees" are added to.

Reply:
nooope. `pooledToken.undistributedLpFees` is decremented as fees are attributed to LPs. 

```
ExchangeRate := (liquidReserves + utilizedReserves - undistributedLpFees) / lpTokenSupply
```

importantly, exchange rate _grows_ as `undistributedLpFees` shrinks. once all undistributed fees have been distributed, `undistributedLpFees` is set to zero.